dev/tools/localization/gen_l10n_types.dart

```diff
@@ -374,19 +377,55 @@ class AppResourceBundle {
     }
 
     String localeString = resources['@@locale'] as String;
-    if (localeString == null) {
-      final RegExp filenameRE = RegExp(r'^[^_]*_(\w+)\.arb$');
-      final RegExpMatch match = filenameRE.firstMatch(file.path);
-      localeString = match == null ? null : match[1];
+
+    // If the locale string was not defined in @@locale, look for the first
+    // instance of an ISO 639-1 language code, matching exactly.
+    final String fileName = path.basenameWithoutExtension(file.path);
+
+    for (int index = 0; index < fileName.length; index += 1) {
+      // If an underscore was found, check if locale string follows.
+      if (fileName[index] == '_' && fileName[index + 1] != null) {
+        // If Locale.tryParse fails, it returns the String 'null'.
```

Comment:
It returns null not "null", right?

Reply:
You're right. In one my previous changes it was returning null because I used `toString()` incorrectly. I'll update the comment